src/nvim/terminal.c

```diff
@@ -205,6 +202,8 @@ void terminal_teardown(void)
 
 Terminal *terminal_open(TerminalOptions opts)
 {
+  // Will this be a true color terminal?
+  bool true_color = os_getenv("NVIM_TUI_ENABLE_TRUE_COLOR") != NULL;
```

Comment:
NVIM_TUI_ENABLE_TRUE_COLOR is a hack that tells the TUI to attach as a "rgb enabled UI". NVIM_TUI_ENABLE_TRUE_COLOR is really not relevant here. The authority is the UI.